src/confluent_kafka/src/Admin.c

```diff
@@ -331,92 +331,87 @@ static int Admin_set_replica_assignment (const char *forApi, void *c_obj,
                         goto err;
                 }
 
-                for (i = 0; i < config_count; i++) {
-                        PyObject *config_op_value, *config_value,
-                                 *op_type;
-                        Py_ssize_t config_value_cnt;
-                        config_op_value = PyList_GET_ITEM(vo, i);
-                        rd_kafka_AlterConfigOpType_t op;
-                        const char *v = NULL;
-
-                        if (!PyList_Check(config_op_value) ||
-                            (config_value_cnt = (int)PyList_Size(config_op_value)) < 2) {
-                                PyErr_SetString(PyExc_ValueError,
-                                                "expected operation type and value");
-                                goto err;
-                        }
+                if (!cfl_PyObject_GetAttr(config_entry, "incremental_operation",
+                                          &incremental_operation, NULL, 1, 0))
+                        goto err;
 
-                        op_type = PyList_GET_ITEM(config_op_value, 0);
-                        config_value = PyList_GET_ITEM(config_op_value, 1);
-                        if (!cfl_PyObject_GetInt(op_type, "value", &op, -1, 1)) {
-                                goto err;
-                        }
+                if (!cfl_PyObject_GetInt(incremental_operation, "value",
+                    &op, -1, 1))
+                        goto err;
 
-                        if (op != RD_KAFKA_ALTER_CONFIG_OP_TYPE_DELETE) {
-                                if (!(vs = cfl_PyObject_Unistr(config_value)) ||
-                                    !(v = cfl_PyUnistr_AsUTF8(vs, &vs8))) {
-                                        PyErr_Format(PyExc_ValueError,
-                                                "expected value name to be unicode "
-                                                "string");
-                                        goto err;
-                                }
-                        }
+                if (!cfl_PyObject_GetString(config_entry, "name", &name, NULL, 1, 0))
+                        goto err;
 
-                        error = rd_kafka_ConfigResource_set_incremental_config(
-                                        (rd_kafka_ConfigResource_t *)c_obj,
-                                        k, op, v);
-                        if (error) {
-                                PyErr_Format(PyExc_ValueError,
-                                        "%s config %s failed: %s",
-                                        op, k, rd_kafka_error_string(error));
-                                rd_kafka_error_destroy(error);
-                                goto err;
-                        }
+                if (op != RD_KAFKA_ALTER_CONFIG_OP_TYPE_DELETE &&
+                    !cfl_PyObject_GetString(config_entry, "value", &value, NULL, 1, 0))
+                        goto err;
```

Comment:
We should let librdkafka handle it for consistent behaviour across all the clients? If not, then we should set proper error here.

Reply:
the error is set by the cfl_PyObject_GetString function there

Reply:
Shall we have a specific error message which suggest that value is necessary for non Delete operations? Also, we can also check empty string here if it makes sense to do that. Your call on both of these.

Reply:
It's ok if you pass an empty string, the value is ignored for delete operations. If I overwrite the error there it could hide the underlying error, that could be missing attribute, value is None or not a unicode string type.